src/features/utils/event-buffer.js

```diff
@@ -25,12 +26,16 @@ export class EventBuffer {
     return this.#buffer.length
   }
 
+  getRequiredVersion () {
+    return this.#buffer.some(({ hasV2Data }) => hasV2Data) ? 2 : 1
+  }
+
   isEmpty () {
     return this.#buffer.length === 0
   }
 
   get () {
-    return this.#buffer
+    return this.#buffer.map(({ event }) => event)
```

Comment:
get should be returning just whatever the buffer has. If there's bad data, then the problem to address is the bad .add